eyeshade/controllers/reports.js

```diff
@@ -55,6 +55,57 @@ v1.publishers = {}
 v2.publishers = {}
 v3.publishers = {}
 
+/*
+   GET /v1/reports/publishers/referrals
+ */
+
+v1.publishers.referrals = {
+  handler: (runtime) => {
+    return async (request, reply) => {
+      const amount = request.query.amount
+      const authority = authorityProvider(request)
+      const currency = request.query.currency.toUpperCase()
+      const reportId = uuid.v4().toLowerCase()
+      const reportURL = url.format(underscore.defaults({ pathname: '/v1/reports/file/' + reportId }, runtime.config.server))
+      const debug = braveHapi.debug(module, request)
+
+      const threshold = runtime.currency.fiat2alt(currency, amount, altcurrency)
+
+      await runtime.queue.send(debug, 'report-publishers-referrals',
+                               underscore.defaults({ reportId: reportId, reportURL: reportURL, authority: authority },
+                                                   { threshold: threshold }, request.query))
+      reply({ reportURL: reportURL })
+    }
+  },
+
+  auth: {
+    strategy: 'session',
+    scope: [ 'ledger', 'QA' ],
+    mode: 'required'
+  },
+
+  description: 'Returns information about referrals to publishers, used to prepare referral payout',
+  tags: [ 'api' ],
+
+  validate: {
+    query: {
+      format: Joi.string().valid('json', 'csv').optional().default('json').description('the format of the report'),
```

Comment:
since you error on requests for 'csv' reports, why not remove the choice of `, 'csv'" ?